src/lib.rs

```diff
@@ -1537,10 +1856,283 @@ mod test {
     }
 
     mod arith {
-        use super::super::{Ratio, Rational};
+        #[cfg(feature = "bigint")]
+        use super::super::BigRational;
+        use super::super::{Ratio, Rational, Rational32, Rational64};
```

Comment:
It's great that you're adding so many tests -- actually impressive that this makes up the bulk of your PR!  However, `src/lib.rs` is getting unwieldy.  Can we move your additions to a new submodule in a separate file?

Reply:
Absolutely. I wanted to do that, but didn't wanna rock the boat too much with my first PR here. :)